### PR TITLE
Uf 3447 fix business rules order

### PR DESCRIPTION
**Pull request by MattiasGradinApendo**
Changed ordet of business rules to check rejections before approval

### CODE REVIEW

File: src/main/java/se/sundsvall/processengine/parkingpermit/businessrules/DriverNewParkingPermitRule.java [hunk 1]
```diff
@@ -52,13 +52,14 @@ public ErrandDecisionWrapper apply(ErrandDecisionWrapper errandDecision) {
 			return errandDecision.withDecision(Optional.empty());
 		}
 
-		// If duration is more than minimum months for approval, create decision based on walking ability
-		if (Period.parse(disabilityDuration).getMonths() >= RULE_MINIMUM_DISABILITY_DURATION_FOR_APPROVAL) {
-			return errandDecision.withDecision(createDecisionBasedOnWalkingAbility(toBoolean(walkingAbility), maxWalkingDistance));
+		// If duration is less than six months
+		if (Period.parse(disabilityDuration).getMonths() == RULE_MINIMUM_DISABILITY_DURATION_FOR_APPROVAL) {
```
Comment (foobar-1337) [inline src/main/java/se/sundsvall/processengine/parkingpermit/businessrules/DriverNewParkingPermitRule.java:56]: `<=` ??

File: src/main/java/se/sundsvall/processengine/parkingpermit/businessrules/PassengerNewParkingPermitRule.java [hunk 2]
```diff
@@ -25,10 +25,10 @@
 @Component
 public class PassengerNewParkingPermitRule implements BusinessRule {
 	private static final Logger LOGGER = LoggerFactory.getLogger(PassengerNewParkingPermitRule.class);
-	private static final String APPROVAL_CANT_BE_ALONE_WHILE_PARKING = "Rekommenderat beslut är bevilja. Sökande kan inte lämnas ensam en kort stund vid parkering.";
+	private static final String APPROVAL_CANT_BE_ALONE_WHILE_PARKING = "Rekommenderat beslut är bevilja. Sökande kan inte lämnas ensam en kort stund vid parkering,kan inte gå mer än 100 meter och funktionsnedsättning är längre än sex månader.";
```
Comment (foobar-1337) [inline src/main/java/se/sundsvall/processengine/parkingpermit/businessrules/PassengerNewParkingPermitRule.java:28]: Add a spece after the ","-character?

File: src/main/java/se/sundsvall/processengine/parkingpermit/businessrules/PassengerNewParkingPermitRule.java [hunk 3]
```diff
@@ -70,13 +75,9 @@ private static boolean validParameters(String disabilityDuration, String canBeAl
 	}
 
 	private static Optional<DecisionDTO> createDecisionBasedOnDisabilityDuration(String disabilityDuration) {
-		if (Period.parse(disabilityDuration).getMonths() >= RULE_MINIMUM_DISABILITY_DURATION_FOR_APPROVAL) {
-			return createDecision(RECOMMENDED, APPROVAL, APPROVAL_CANT_BE_ALONE_WHILE_PARKING);
+		if (Period.parse(disabilityDuration).getMonths() == RULE_MINIMUM_DISABILITY_DURATION_FOR_APPROVAL) {
```
Comment (foobar-1337) [inline src/main/java/se/sundsvall/processengine/parkingpermit/businessrules/PassengerNewParkingPermitRule.java:78]: Same as comment above:

`<=` ??

File: src/main/java/se/sundsvall/processengine/parkingpermit/businessrules/DriverNewParkingPermitRule.java (hunk 1 above)
Comment (MattiasGradinApendo) [reply src/main/java/se/sundsvall/processengine/parkingpermit/businessrules/DriverNewParkingPermitRule.java:56]: This needs a comment I realise. The logic is that we use Periods as some kind of enums, see https://sundsvall.atlassian.net/wiki/spaces/SKA/pages/1034256415/CaseData#Mappningstabeller . So P6M or six months means less than six months. Does it make sense? Not really, but it needs to be changed in CaseData.

File: src/main/java/se/sundsvall/processengine/parkingpermit/businessrules/PassengerNewParkingPermitRule.java (hunk 2 above)
Comment (MattiasGradinApendo) [reply src/main/java/se/sundsvall/processengine/parkingpermit/businessrules/PassengerNewParkingPermitRule.java:28]: Will fix

File: src/main/java/se/sundsvall/processengine/parkingpermit/businessrules/DriverNewParkingPermitRule.java (hunk 1 above)
Comment (MattiasGradinApendo) [reply src/main/java/se/sundsvall/processengine/parkingpermit/businessrules/DriverNewParkingPermitRule.java:56]: Yes will do that :)

**Comment by MattiasGradinApendo** [comment]
Will close this and a new PR will be made from a different branch, these changes are not neccecary as they are already built correctly in UF-3345-handle-updates-process-engine

**Pull request closed without merging**